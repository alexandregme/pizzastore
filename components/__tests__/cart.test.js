import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import Cart from '../cart';
import CartItem from "../cart-item";

describe("Cart", ()=>{
  let mountedCart;

  beforeEach(()=>{
    mountedCart = shallow(<Cart />);
  });

  it('renders without crashing', () => {
    shallow(<Cart />);
    expect(mountedCart).toMatchSnapshot();
  });

  it('expect to show no empty cart msg', () => {
    let emptyCart = mountedCart.find('span');
    expect(emptyCart).toHaveLength(1);
    expect(emptyCart.text()).toEqual('your cart is empty.');
  });
});

describe("Cart Custom component props", ()=>{
  let mountedCart;
  let props = {
    cartItems: ['SMALL', 'MEDIUM', 'LARGE']
  };

  beforeEach(()=>{
    mountedCart = shallow(<Cart {...props} />);
  });

  it('expect to render 3 items in the cart', () => {
    let cart = mountedCart.find(CartItem);
    expect(cart.length).toEqual(3);
  });
});
