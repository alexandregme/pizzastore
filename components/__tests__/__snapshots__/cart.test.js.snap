// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cart />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <span>
        your cart is empty.
      </span>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "your cart is empty.",
      },
      "ref": null,
      "rendered": "your cart is empty.",
      "type": "span",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <span>
          your cart is empty.
        </span>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "your cart is empty.",
        },
        "ref": null,
        "rendered": "your cart is empty.",
        "type": "span",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
